src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -1275,9 +1313,97 @@ bool entitiesQuery
+            needToAdd = false;
+            break;  /* jx */
+          }
+        }
+        if (needToAdd)
+        {
+          ContextElementResponse* cerP = new ContextElementResponse();
+          cerP->contextElement.entityId.id = enV.get(ix)->id;
+          cerP->contextElement.entityId.type = enV.get(ix)->type;
+          cerP->contextElement.entityId.isPattern = "false";
+
+          /* This entity has to be prunned if after CPr searching no attribute is "added" to it. The notPrune attribute distinguish this kind of
+           * entities from the "real ones" that are without attributes in the Orion local database (a rare case, but may happen) */
+          cerP->notPrune = false;
+
+          for (unsigned int jx = 0; jx < attrL.size(); ++jx)
+          {
+            ContextAttribute* caP = new ContextAttribute(attrL.get(jx), "", "");
+            caP->found = false;
+            cerP->contextElement.contextAttributeVector.push_back(caP);
+          }
+
+          cerP->statusCode.fill(SccOk);
+
+          cerV->push_back(cerP);
+        }
+      }
+    }
+
     return true;
 }
 
+/* ****************************************************************************
+*
+* pruneNotFoundContextElements -
+*
+* Remove attributes in the vector with 'found' value is 'false'
+*
+*/
+void pruneNotFoundContextElements(ContextElementResponseVector& oldCerV, ContextElementResponseVector* newCerVP)
+{
+  for (unsigned int ix = 0; ix < oldCerV.size(); ++ix)
+  {
+    ContextElementResponse* cerP = oldCerV.get(ix);
+    ContextElementResponse* newCerP = new ContextElementResponse();
+
+    /* Note we cannot use the ContextElement::fill() method, given that it also copies the ContextAttributeVector. The side-effect
+     * of this is that attributeDomainName and domainMetadataVector and not being copied, but it should not be a problem, given that
+     * domain attributes are not implemented */
+    newCerP->contextElement.entityId.fill(&cerP->contextElement.entityId);
+    newCerP->contextElement.providingApplicationList = cerP->contextElement.providingApplicationList;  // FIXME P10: not sure if this is the right way of doing, maybe we need a fill() method for this
+    newCerP->statusCode.fill(&cerP->statusCode);
+
+    bool notPruneEntity = cerP->notPrune;
+
+    for (unsigned int jx = 0; jx < cerP->contextElement.contextAttributeVector.size(); ++jx)
+    {
+      ContextAttribute* caP = cerP->contextElement.contextAttributeVector.get(jx);
+      if (caP->found)
+      {
+        ContextAttribute* newCaP = new ContextAttribute(caP);
+        newCerP->contextElement.contextAttributeVector.push_back(newCaP);
+      }
+
+    }
+
+    /* If after prunning the entity has no attribute and no CPr information, then it is not included
+     * in the output vector, except if "notPrune" is set to true */
+    if (!notPruneEntity && newCerP->contextElement.contextAttributeVector.size() == 0 && newCerP->contextElement.providingApplicationList.size() == 0)
```

Comment:
I just hate double negations!
Might be a problem with my brain, I don't know.

So much easier to read
    if (pruneEntity)
than
    if (!notPruneEntity)

[ just my opinion ]

NTC